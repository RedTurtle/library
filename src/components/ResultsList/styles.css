.results-list {
    margin: 0 auto;
    padding: 15px;
}

.result-item {
    margin-bottom: 4rem;
}

.results-list h3 {
    padding-bottom: 0.6rem;
    border-bottom: 1px solid #bbb;
}

.results-list a {
    color: #d40e0e;
}

.results-list a:hover,
.results-list a:active {
    color: #980a0a;
}

@media (min-width: 900px) {
    .results-list {
        width: 900px;
    }
}
