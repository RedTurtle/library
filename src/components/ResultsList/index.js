// @flow
import React from 'react';
import './styles.css';
import typeof data from '../../data';

type Props = { results: data };

const ResultsList = ({ results }: Props) => (
  <div className="results-list">
    {results.map(item => (
      <div key={item.id} className="result-item">
        <h3>{item.name}</h3>
        <p>
          <a href={item.url}>{item.url}</a>
        </p>
        {item.keywords &&
        item.keywords.length && <p>Keywords: {item.keywords.join(', ')}</p>}
      </div>
    ))}
    {!results.length && <p>No results matched your criteria.</p>}
  </div>
);

export default ResultsList;
