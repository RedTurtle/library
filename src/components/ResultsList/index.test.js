import React from 'react';
import ResultsList from './index';
import { shallow } from 'enzyme';

const results = [
  {
    id: 'collective.tiles.advancedstatic',
    name: 'collective.tiles.advancedstatic',
    url: 'https://github.com/collective/collective.tiles.advancedstatic',
    keywords: ['plone', 'plone5', 'collective', 'static', 'tiles', 'advanced'],
  },
  {
    id: 'collective.tiles.collection',
    name: 'collective.tiles.collection',
    url: 'https://github.com/collective/collective.tiles.collection',
    keywords: ['plone', 'plone5', 'collective', 'collection', 'tiles'],
  },
];

it('matches previous snapshot', () => {
  expect(shallow(<ResultsList results={results} />)).toMatchSnapshot();
});
