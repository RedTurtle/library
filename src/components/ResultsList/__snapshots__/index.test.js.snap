// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches previous snapshot 1`] = `
<div
  className="results-list"
>
  <div
    className="result-item"
  >
    <h3>
      collective.tiles.advancedstatic
    </h3>
    <p>
      <a
        href="https://github.com/collective/collective.tiles.advancedstatic"
      >
        https://github.com/collective/collective.tiles.advancedstatic
      </a>
    </p>
    <p>
      Keywords: 
      plone, plone5, collective, static, tiles, advanced
    </p>
  </div>
  <div
    className="result-item"
  >
    <h3>
      collective.tiles.collection
    </h3>
    <p>
      <a
        href="https://github.com/collective/collective.tiles.collection"
      >
        https://github.com/collective/collective.tiles.collection
      </a>
    </p>
    <p>
      Keywords: 
      plone, plone5, collective, collection, tiles
    </p>
  </div>
</div>
`;
